### ISSUE TITLE
"Make chaff" closes the application

### ISSUE THREAD

**Issue by az0** [bleachbit/bleachbit#1444]
To reproduce

1. Open application (either with admin permissions or not)
2. Click the application menu
3. Choose the *Make Chaff* option

What happens

- Application immediately closes
- No error message in GUI
- Nothing printed to console (bleachbit_console.exe)

System

- Build 2310 (new Python 3.10)
- Windows 10

**Comment by sanjaykaria** [comment]
Apparently this issue occurs in Build 2310 and not in the corresponding source branch or in the prior build 2275. It seems that the dependent files for "Make Chaff" are missing in Build 2310.

**Comment by az0** [comment]
Yes, I see that [in the log](https://ci.appveyor.com/project/az0/bleachbit/builds/46017747?fullLog=true#L2143) (with context):

```
x86-windows\tools\python3\..\..\tools\glib\gspawn-win32-helper.exe
1 File(s) copied
g_module_open() failed for C:\projects\bleachbit\x86-windows\tools\python3\pixbufloader-svg.dll: 'C:\projects\bleachbit\x86-windows\tools\python3\pixbufloader-svg.dll': The specified module could not be found.
if (-not (Test-Path upx.zip)) { Start-FileDownload "https://github.com/upx/upx/releases/download/v3.96/upx-3.96-win64.zip" -FileName upx.zip}
Downloading upx.zip (468,845 bytes)...100%
```

A few lines above the file is copied

```
x86-windows\tools\python3\..\..\bin\pixbufloader-svg.dll
1 File(s) copied
```

The file pixbufloader-svg.dll is included in the portable zip file with the size 10240 bytes.

The error seems to come from the `windows/python-gtk3-fixup.bat` file and this line

```bat
%gtk3dir%\tools\gdk-pixbuf\gdk-pixbuf-query-loaders --update-cache %PYTHON_HOME%\pixbufloader-svg.dll
```

My guess is that solution for the pixbuf error is to make sure the dll is in the path when running this command.

This command looks like it is to build cache for loading images, such as icons, which generally work as seen on the application toolbar, so why do you think this error is related to the "make chaff" crash? @sanjaykaria

**Comment by sanjaykaria** [comment]
I have followed all the steps in the appveyor.xml on my local machine and the output on the Appveyor build machine and my local matches. The error starts from the line you have highlighted, as below 

`%gtk3dir%\tools\gdk-pixbuf\gdk-pixbuf-query-loaders --update-cache %PYTHON_HOME%\pixbufloader-svg.dll`

The file pixbufloader-svg.dll is in place on my local machine but a dependency of pixbufloader-svg.dll i.e. kernel32.dll is missing a dependent DLL "ext-ms-win-oobe-query-l1-1-0.dll", as per attached screenshot.
![image](https://user-images.githubusercontent.com/3830310/214491000-282adc6b-7602-460c-99bf-5cf70cf1d04f.png)

I have seen this exact error while running BB from source, very initially, but do not recall how exactly it got resolved. Although I am sure that I had not copied/replaced the missing DLL, highlighted in the screenshot.

**Comment by sanjaykaria** [comment]
> This command looks like it is to build cache for loading images, such as icons, which generally work as seen on the application toolbar, so why do you think this error is related to the "make chaff" crash? @sanjaykaria

@az0 
When I debug Bleachbit, and click the "Make Chaff" button, the app crashes with the below message in the debugger -

```
(bleachbit.py:1084): Gtk-WARNING **: 11:13:14.731: Could not load a pixbuf from /org/gtk/libgtk/theme/Adwaita/assets/check-symbolic.svg.
This may indicate that pixbuf loaders or the mime database could not be found.

(bleachbit.py:1084): GLib-GIO-ERROR **: 11:13:22.401: No GSettings schemas are installed on the system
```

This error seems to be related to the above error of building the image caches, I think

**Comment by sanjaykaria** [comment]
> This command looks like it is to build cache for loading images, such as icons, which generally work as seen on the application toolbar, so why do you think this error is related to the "make chaff" crash? @sanjaykaria

When the "Make Chaff" is invoked, BB tries to load the image for the destination folder button from GTK. So, if we remove/comment the below lines commented from GuiChaff.py, the crash will not occur. But this is obviously not the solution, as the issue seems to be with loading the images

```
        folder_box = Gtk.Box(orientation=Gtk.Orientation.HORIZONTAL)
        folder_box.add(Gtk.Label(_("Select destination folder")))
        self.choose_folder_button = Gtk.FileChooserButton()
        # self.choose_folder_button.set_action(
        #     Gtk.FileChooserAction.SELECT_FOLDER)
        import tempfile
        self.choose_folder_button.set_filename(tempfile.gettempdir())
        folder_box.add(self.choose_folder_button)
        box.add(folder_box)
```

Related to this issue : https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/issues/141

**Comment by sanjaykaria** [comment]
@az0

It seems that this [issue ]( https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/issues/141)is open from quite some time and there might be more similar issues.

However the gdk-pixbuf issue does not occur when running BleachBit with GTK3 binaries from Mysys2.

Is using GTK3 binaries from Mysys2 for the BB build an option?

**Comment by az0** [comment]
> Is using GTK3 binaries from Mysys2 for the BB build an option?

@mkhon 

Would this be a good solution for this issue?

**Comment by az0** [comment]
@mkhon 
Please take a look

**Comment by az0** [comment]
tl;dr It still crashes.

With AppVeyor build 4.6.2.2691 of commit 8df6fd87fedb521681364cbe9b291cb06e13faf4 (Python 3.10.7) the following message is printed to console (when using bleachbit_console.exe) during the crash

```(org.bleachbit.BleachBit:9612): GLib-GIO-ERROR **: 21:24:30.073: No GSettings schemas are installed on the system```

Earlier on startup it states

```(org.bleachbit.BleachBit:12180): GLib-GIO-WARNING **: 21:26:05.993: win32 session dbus binary not found```

Tip: to see the first message, open a cmd.exe window, run bleachbit_console.exe, and answer "no"  to "do you want to allow this app... to make changes".

**Comment by az0** [comment]
`x86-windows\tools\glib\glib-compile-schemas.exe x86-windows\share\glib-2.0\schemas` makes `x86-windows\share\glib-2.0\schemas\gschemas.compiled`

**Comment by az0** [comment]
Fixed in CI build 4.6.2.2697 / commit 0c6d316b7d3e7848b1baa89159750d36e2495baf as tested on Microsoft Windows 11 as both portable and installed
